x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/RestCursorState.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.sql.plugin;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.core.Tuple;
+import org.elasticsearch.xpack.sql.SqlIllegalArgumentException;
+import org.elasticsearch.xpack.sql.action.BasicFormatter;
+import org.elasticsearch.xpack.sql.common.io.SqlStreamInput;
+import org.elasticsearch.xpack.sql.common.io.SqlStreamOutput;
+
+import java.io.IOException;
+import java.time.ZoneOffset;
+import java.util.List;
+
+import static org.elasticsearch.core.Tuple.tuple;
+
+public record RestCursorState(BasicFormatter formatter) implements Writeable {
```

Comment:
Not sure wether this class is a bit of an overkill right now but I added it for additional clarity and to organize the state related code in one place. Also, it could act as an extension point in the future in case we have to add additional state for other formatters. Alternatively, we could just pass `BasicFormatter` everywhere.